internal/internal_workers_test.go

```diff
@@ -331,6 +331,134 @@ func (s *WorkersTestSuite) TestLongRunningDecisionTask() {
 		}
 	}).Times(2)
 
+	options := WorkerOptions{
+		Logger:                zap.NewNop(),
+		DisableActivityWorker: true,
+		Identity:              "test-worker-identity",
+	}
+	worker := newAggregatedWorker(s.service, domain, taskList, options)
+	worker.Start()
+	// wait for test to complete
+	select {
+	case <-doneCh:
+		break
+	case <-time.After(time.Second * 4):
+	}
+	worker.Stop()
+
+	s.True(isWorkflowCompleted)
+	s.Equal(2, localActivityCalledCount)
+}
+
+func (s *WorkersTestSuite) TestMultipleLocalActivities() {
```

Comment:
I don't see how it tests the race condition with the query. May be use the newly added integration tests that use Cadence docker?

Reply:
This does not test the race condition (and technically does not add coverage), but is a better exercising of the local activity logic than just waiting for one.. 

I have not been able to get reliable repro for that - but will try adding a test with docker where I can force that scenario..